implementations/rust/ockam/ockam_transport_core/Cargo.toml

```diff
@@ -14,22 +14,35 @@ Generic Transport primitives.
 """
 publish = true
 rust-version = "1.56.0"
+resolver = "2"
 
 [features]
 default = ["std"]
 
 # Feature (enabled by default): "std" enables functionality expected to
 # be available on a standard platform.
-std = ["ockam_core/std"]
+std = [
+    "ockam_core/std",
+    "ockam_node/std",
+    "tokio",
+    "tokio/io-util",
+    "tokio/net",
+]
```

Comment:
1. I think this is not a problem since `tokio` is an optional dependency that's only included with `std` enabled. I only use `tokio` here for the implementations of the `AsyncRead` and `AsyncWrite` traits on `std`. I could completely remove the dependency by using the newtype pattern to do the implementation in `ockam_transport_tcp` if that's preferable.
2.  I figured that we can start extracting the behavior as the weirdness appears, that way we maximize the shared behavior.
3. Although not particularly a big change while working on this the `heartbeat` was added and by solving conflicts it was automatically enjoyed by the smoltcp crate.

All in all I thought it was a good idea as the behavior they share don't seem coincidental and maybe in the future more protocols could use this worker/router implementation.

Anyways, I understand the points and also the generic implementation along with all the trait bounds make the code much harder to parse so if you still think is better to have each crate with their own worker/router implementation I will go back to that!

Reply:
Valid points all but I do feel like the end result is a bigger increase in overall code complexity than what we're gaining through de-duplication here.

Thoughts @thomcc @SanjoDeundiak @spacekookie ?